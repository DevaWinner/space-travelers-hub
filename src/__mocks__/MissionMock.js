import React from 'react';
import { useSelector } from 'react-redux';

const MissionMock = () => {
  const missions = useSelector((state) => state.missions.missions);
  const filterMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div className="missions-wrap">
      <h2>My Mission</h2>
      <table>
        <tbody>
          {filterMissions.map((mission) => (
            <tr key={mission.id}>
              <td>{mission.name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default MissionMock;
