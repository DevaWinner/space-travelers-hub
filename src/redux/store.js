import rocketsSlice from './rockets/rocketsSlice';
