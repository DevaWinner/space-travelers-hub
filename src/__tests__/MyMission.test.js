import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import MissionMock from '../__mocks__/MissionMock';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('MissionMock', () => {
  it('render My Missions Components', () => {
    render(<MissionMock />);
    expect(MissionMock).toMatchSnapshot();
  });
});
