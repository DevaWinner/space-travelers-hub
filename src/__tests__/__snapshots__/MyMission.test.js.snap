// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionMock render My Missions Components 1`] = `[Function]`;
